src/autoscaler/metricsforwarder/server/server.go

```diff
@@ -17,31 +17,84 @@ import (
 	}
 
-	mh := NewCustomMetricsHandler(logger, metricForwarder, policyDB, allowedMetricCache)
+	mh := NewCustomMetricsHandler(logger, *metricForwarder, policyDb, allowedMetricCache)
 	authenticator, err := auth.New(logger, credentials)
 	if err != nil {
-		return nil, fmt.Errorf("failed to add auth middleware: %w", err)
+		return nil, fmt.Errorf("failed to create auth middleware: %w", err)
+	}
+
+	mfRouter := setupMetricForwarderRouter(logger, mh, authenticator, httpStatusCollector, rateLimiter)
+	healthRouter, err := createHealthRouter(policyDb, credentials, logger, conf, httpStatusCollector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create health router: %w", err)
+	}
+
+	mainRouter := setupMainRouter(mfRouter, healthRouter)
+	return helpers.NewHTTPServer(logger, conf.Server, mainRouter)
+}
+
+func createMetricForwarder(logger lager.Logger, conf *config.Config) (*forwarder.MetricForwarder, error) {
+	metricForwarder, err := forwarder.NewMetricForwarder(logger, conf)
+	if err != nil {
+		logger.Error("failed-to-create-metricforwarder-server", err)
+		return nil, err
 	}
+	return &metricForwarder, nil
+}
+
+func setupMetricForwarderRouter(logger lager.Logger, mh *CustomMetricsHandler, authenticator *auth.Auth, httpStatusCollector healthendpoint.HTTPStatusCollector, rateLimiter ratelimiter.Limiter) *mux.Router {
 	httpStatusCollectMiddleware := healthendpoint.NewHTTPStatusCollectMiddleware(httpStatusCollector)
 	rateLimiterMiddleware := ratelimiter.NewRateLimiterMiddleware("appid", rateLimiter, logger.Session("metricforwarder-ratelimiter-middleware"))
 
-	r := routes.MetricsForwarderRoutes()
-	r.Use(otelmux.Middleware("metricsforwarder"))
-	r.Use(rateLimiterMiddleware.CheckRateLimit)
-	r.Use(httpStatusCollectMiddleware.Collect)
-	r.Use(authenticator.Authenticate)
-	r.Get(routes.PostCustomMetricsRouteName).Handler(common.VarsFunc(mh.VerifyCredentialsAndPublishMetrics))
+	mfRouter := routes.MetricsForwarderRoutes()
+	mfRouter.Use(otelmux.Middleware("metricsforwarder"))
+	mfRouter.Use(rateLimiterMiddleware.CheckRateLimit)
+	mfRouter.Use(httpStatusCollectMiddleware.Collect)
+	mfRouter.Use(authenticator.Authenticate)
+	mfRouter.Get(routes.PostCustomMetricsRouteName).Handler(common.VarsFunc(mh.VerifyCredentialsAndPublishMetrics))
+
+	return mfRouter
+}
+
+func createHealthRouter(policyDb db.PolicyDB, credDb cred_helper.Credentials, logger lager.Logger, conf *config.Config, httpStatusCollector healthendpoint.HTTPStatusCollector) (*mux.Router, error) {
+	checkers := []healthendpoint.Checker{
+		healthendpoint.DbChecker(db.PolicyDb, policyDb),
+		healthendpoint.DbChecker(db.StoredProcedureDb, credDb),
+	}
+
+	gatherer := createPrometheusRegistry(policyDb, httpStatusCollector, logger)
+	healthRouter, err := healthendpoint.NewHealthRouter(conf.Health, checkers, logger.Session("health-server"), gatherer, time.Now)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create health router: %w", err)
```

Comment:
This does not add any valuable information compared e.g. to line 41. What about this:
```suggestion
		return nil, err
```